marconi-tutorial/src/Marconi/Tutorial/Indexers/AddressCount.hs

```diff
@@ -54,36 +59,50 @@ import Marconi.Core.Experiment (
   point,
  )
 import Marconi.Core.Experiment qualified as Core
+import Network.JsonRpc.Types (JsonRpcErr)
+
+type SQLiteStandardIndexer event = Core.StandardIndexer IO Core.SQLiteIndexer event
+type AddressCountIndexer = SQLiteStandardIndexer AddressCountEvent
 
 addressCountWorker
   :: FilePath
   -> SecurityParam
   -> IO
-      ( Core.WorkerIndexer
+      ( Core.StandardWorker
           IO
-          (WithDistance BlockEvent)
+          BlockEvent
           AddressCountEvent
-          (Core.MixedIndexer Core.SQLiteIndexer Core.ListIndexer)
+          Core.SQLiteIndexer
       )
 addressCountWorker dbPath securityParam = do
   let extract = getEventsFromBlock
-  ix <- Utils.toException $ mkAddressCountMixedIndexer dbPath securityParam
-  Core.createWorker "AddressCount" extract ix
+  -- ix <- Utils.toException $ mkAddressCountMixedIndexer dbPath securityParam
+  -- Core.createWorker "AddressCount" extract ix
+  ix <- Utils.toException $ mkAddressCountSqliteIndexer dbPath
+  let config =
+        Core.StandardWorkerConfig
+          "AddressCount"
+          securityParam
+          (Core.CatchupConfig 2_000 10)
+          (pure . extract)
+          nullTracer
+  Core.mkStandardWorker config ix
 
 newtype AddressCountEvent = AddressCountEvent {unAddressCountEvent :: Map C.AddressAny Int}
   deriving (Show)
 
 type instance Point AddressCountEvent = C.ChainPoint
 
 newtype AddressCountQuery = AddressCountQuery C.AddressAny
-  deriving (Eq, Ord, Show)
+  deriving stock (Eq, Ord, Show, Generic)
+  deriving anyclass (FromJSON)
 
 type instance Result AddressCountQuery = Int
 
 -- Note that for Marconi, it is more optimal to return `Nothing` instead of
 -- `Just $ AddressCountEvent mempty`.
-getEventsFromBlock :: WithDistance BlockEvent -> Maybe AddressCountEvent
-getEventsFromBlock (WithDistance _ (BlockEvent (C.BlockInMode (C.Block _ txs) _) _ _)) =
+getEventsFromBlock :: BlockEvent -> Maybe AddressCountEvent
+getEventsFromBlock (BlockEvent (C.BlockInMode (C.Block _ txs) _) _ _) =
```

Comment:
I'm not a huge fan of this pattern matching, but I can't provide a nice alternative.

Reply:
Perhaps a combination of view patterns and lenses?

Reply:
Maybe extend `cardano-api-extended` (no pun intended) with accessors for the fields of `BlockInMode`?

Reply:
I'm not a huge fan of patterns (I'm probably too old school to like them) but it's a personal preference, so I wouldn't interfere if you think it's a good idea. Putting it, or some standard accessors or lenses in `cardano-api-extended` sounds like a good idea.

Reply:
There are pattern synonyms (defined with the `pattern` keyword) and view patterns (used with `->` to pre-apply a function to a binding. Do you dislike both, @berewt, or just one of them?

Reply:
Both, the first because it adds complexity to the syntax for few benefits, the second because it obfuscates the original data structure without giving you any hint to notice it. It may look like a type has several constructors, while you only look at one constructor and some patterns.
(again, it's highly subjective)

Reply:
I was thinking of something like this:
```suggestion
getEventsFromBlock (preview blockTxs -> Just txs) =
```
where `blockTxs` is a lens.

Reply:
I've just discovered that the `C.Block` constructor is itself a pattern! 😂

```haskell
pattern Block :: BlockHeader -> [Tx era] -> Block era
pattern Block header txs <- (getBlockHeaderAndTxs -> (header, txs))
```

Reply:
On your proposal, I would advocate that I don't see a major improvement over `getEventsFromBlock bim = let txs = view blockTxs bim in ...` (but we can easily use one or the other, based on personal taste, which is totally fine).
And yes `C.Block` is a pattern, guess which pattern reminds me that I don't like them... :)

Reply:
I was thinking that this was one of several equations for the function, and therefore needs to be a guard (hence `preview` rather than `view`) but I see now that it's the only one. In this case, the `let` binding is probably clearer.

However, I noticed that the function currently uses both `let` and `where`. I think it's better to use one or the other but not both in any single function.